src/components/Header.tsx

```diff
@@ -1,20 +1,128 @@
+  const [isOpen, setIsOpen] = useState(false);
+
+  useEffect(() => {
+    if (!isOpen) return;
+    function handleRouteChange() {
+      setIsOpen(false);
+    }
+    Router.events.on('routeChangeComplete', handleRouteChange);
+    return () => {
+      Router.events.off('routeChangeComplete', handleRouteChange);
+    };
+  }, [isOpen]);
 
-export const Header: React.FC<HeaderProps> = ({ navigation }) => {
+  return (
+    <div className={clsx(className, display)} {...props}>
+      <button
+        type="button"
+        className="flex h-8 w-8 items-center justify-center text-slate-500 hover:text-slate-600 dark:text-slate-400 dark:hover:text-slate-300"
+        onClick={() => setIsOpen(true)}
+      >
+        <span className="sr-only">Navigation</span>
+        <svg width="24" height="24" fill="none" aria-hidden="true">
+          <path
+            d="M12 6v.01M12 12v.01M12 18v.01M12 7a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm0 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm0 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z"
+            stroke="currentColor"
+            strokeWidth="1.5"
+            strokeLinecap="round"
+            strokeLinejoin="round"
+          />
+        </svg>
+      </button>
+      <Dialog
+        as="div"
+        className={clsx('fixed inset-0 z-50', display)}
+        open={isOpen}
+        onClose={setIsOpen}
+      >
+        <Dialog.Overlay className="fixed inset-0 bg-black/20 backdrop-blur-sm dark:bg-slate-900/80" />
+
+        <div className="dark:highlight-white/5 fixed top-4 right-4 w-full max-w-xs rounded-lg bg-white p-6 text-base font-semibold text-slate-900 shadow-lg dark:bg-slate-800 dark:text-slate-400">
+          <button
+            type="button"
+            className="absolute top-5 right-5 flex h-8 w-8 items-center justify-center text-slate-500 hover:text-slate-600 dark:text-slate-400 dark:hover:text-slate-300"
+            onClick={() => setIsOpen(false)}
+          >
+            <span className="sr-only">Close navigation</span>
+            <svg
+              viewBox="0 0 10 10"
+              className="h-2.5 w-2.5 overflow-visible"
+              aria-hidden="true"
+            >
+              <path
+                d="M0 0L10 10M10 0L0 10"
+                fill="none"
+                stroke="currentColor"
+                strokeWidth="2"
+                strokeLinecap="round"
+              />
+            </svg>
+          </button>
+          <ul className="space-y-6">
+            <NavItems />
+            <li>
+              <a
+                href="https://github.com/codealoy/codealoy"
+                className="hover:text-sky-500 dark:hover:text-sky-400"
+              >
+                GitHub
+              </a>
+            </li>
+          </ul>
+          <div className="mt-6 flex justify-between border-t border-slate-200 pt-6 dark:border-slate-200/10">
+            <span> Select Theme</span> <ThemeSelector />
+          </div>
+        </div>
+      </Dialog>
+    </div>
+  );
+}
+export function NavItems() {
```

Comment:
`NavItems` should be in a separate component file.

Reply:
* Created a separate component